experimental/packages/opentelemetry-sdk-metrics-base/src/InstrumentDescriptor.ts

```diff
@@ -46,3 +46,35 @@ export function createInstrumentDescriptorWithView(view: View, instrument: Instr
     valueType: instrument.valueType,
   };
 }
+
+export function isDescriptorCompatibleWith(descriptor: InstrumentDescriptor, otherDescriptor: InstrumentDescriptor) {
+  return descriptor.name === otherDescriptor.name
+    && descriptor.description === otherDescriptor.description
```

Comment:
> Note the `description` property of an instrument is explicitly
disregarded when considering duplicate registration, because it is not
semantic in nature.  Implementations SHOULD use any of the provided
`description` values when emitting metrics that had duplicate registrations.
> https://github.com/open-telemetry/opentelemetry-specification/pull/2270

`description` should not be considered as part of the descriptor comparisons.

Reply:
I removed description here as the proposed version of the spec suggests to use the longer description instead for this type of conflict. I'm resolving this conversation for now.